### PR TITLE
06314 - Platform Add/Remove Node Integration Tests

### LINKED ISSUE #6314: Noncontiguous NodeId - Integration Tests

### LINKED ISSUE #7341: Noncontiguous NodeId - Implement Integration Tests

### PR DESCRIPTION
**Pull request by edward-swirldslabs**
This PR creates two new AddressBookTestingTool test scenarios.  One for adding a single node to the network.  One for removing a single node from the network.  

This PR also modifies the name of the file which records the address book used at start up to append the node creating the file.  When run locally, multiple nodes generate the same file and this was masking difference in the file for added nodes.  

**Description**:
Fixes #6314 
Fixes #7341

**Needs To Be Merged After** 
https://github.com/swirlds/swirlds-platform-regression/pull/3731

### CODE REVIEW

**Review by lpetrovic05 — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookInitializer.java [hunk 1]
```diff
@@ -105,11 +108,13 @@ public class AddressBookInitializer {
      * @param platformContext   The context for the platform.
      */
     public AddressBookInitializer(
+            @NonNull final NodeId selfId,
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookInitializer.java:111]: Javadoc for new param

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java [hunk 2]
```diff
@@ -1087,6 +1088,17 @@ private AddressBook getAddressBook() {
         return addressBook;
     }
 
+    /**
+     * Checks if the event has the same index and the one provided.
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java:1092]: Was this meant to say "_as_ the one provided" ?

That being said, I don't think this doc is a great description of what's being checked. The method is checking whether the *creator* of the event has a certain index in the address book. I think the parameter docs could also be updated for clarity

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java [hunk 3]
```diff
@@ -1087,6 +1088,17 @@ private AddressBook getAddressBook() {
...
+     *
+     * @param e the event to check for euqality to index
+     * @param index the index to check for equality to the event's index
+     * @return true if the event has the same index as the one provided, false otherwise
+     */
+    private boolean hasIndex(@NonNull final EventImpl e, final int index) {
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java:1098]: I think this method could have a better name, `hasIndex` isn't very informative IMO

What do you think of `wasEventCreatedByNodeIndex`? Or maybe you can come up with a better alternative

Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java:1098]: Nit: I'm not a fan of single letter variable names. I'd personally change it to `event` (even though the rest of the file uses single letters). Up to you, though

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java [hunk 4]
```diff
@@ -1087,6 +1088,17 @@ private AddressBook getAddressBook() {
         return addressBook;
     }
 
+    /**
+     * Checks if the event has the same index and the one provided.
+     *
+     * @param e the event to check for euqality to index
```
Comment (alittley) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java:1094]: typo "euqality"

**Review by cody-littley — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookInitializer.java (hunk 1 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookInitializer.java:111]: adding

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java (hunk 2 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java:1092]: updated javadoc for clarity.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java (hunk 3 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java:1098]: changed to `eventCreatorIndexMatches`

Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java:1098]: changed to `event`

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java (hunk 4 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ConsensusImpl.java:1094]: fixed typo.

**Review by cody-littley — APPROVED** [review]

**Review by JeffreyDallas — APPROVED** [review]

**Pull request merged**